sleuth_backend/solr/connection.py

```diff
@@ -0,0 +1,96 @@
+import pysolr
+import json
+
+class SolrConnection(object):
+    """
+    Connection to Solr database
+    """
+    
+    def __init__(self, url):
+        """
+        Creates a SolrConnection form the given base Solr url of the form
+        'http://solrhostname:solrport/solr'.
+        """
+        self.url = url
+        self.solr = pysolr.Solr(url, timeout=10)
+        self.solr_admin = pysolr.SolrCoreAdmin(url + '/admin/cores')  
+        self.cores = {}
+
+        for core_name in self.fetch_core_names():
+            self.cores[core_name] = pysolr.Solr(self.url + '/' + core_name)
+
+    def fetch_core_names(self):
+        """
+        Makes a request to Solr and returns an array of strings where each 
+        string is the name of a core in the response from Solr.
+        """
+        status_response = self.solr_admin.status()
+        status = json.loads(status_response)
+        return [core_name for core_name in status['status']]
+
+    def core_names(self):
+        """
+        Returns a list of know cores in the Solr instance without making a 
+        request to Solr.
+        """
+        return list(self.cores.keys())
+
+    def fetch_core_schema(self, name):
+        """
+        Returns the schema of the core with the given name as a dictionary.
+        """
+        response = self.solr_admin._get_url(self.url + '/' + name + '/schema')
+        body = json.loads(response)
+
+        if 'schema' not in body:
+            raise ValueError('Solr did not return a schema. Are you sure the core' + 
+                ' name {} is an existing core?'.format(name))
+        
+        return body['schema']
+
+    def insert_document(self, core, doc):
```

Comment:
An idea, we might want something to handle inserting an item that is already present in the core (aka updating an existing item)